binder/src/main/java/io/grpc/binder/internal/BinderServer.java

```diff
@@ -129,10 +135,15 @@ public synchronized void shutdown() {
       // Break the connection to the binder. We'll receive no more transactions.
       hostServiceBinder.setHandler(GoAwayHandler.INSTANCE);
       listener.serverShutdown();
+      // TODO(jdcormie): Shouldn't this happen in onTerminated()? Is this even used anywhere?
```

Comment:
It looks like BinderServerTransport uses the pool, so the reference here is probably to just keep it alive and prevent creating/destroying it repeatedly if there's only one transport at a time.

It would probably be fine to be in onTerminated, but it also seems fine here. The transports are still using it, but they have their own reference.

Reply:
Yeah I was confused because each BinderServerTransport checks out an Executor from that pool for itself. No mention of an anti-thrashing rationale in https://github.com/grpc/grpc-java/pull/8210 where the pool was added (but the executorService member was preexisting). I wonder if it was just an oversight. Wouldn't the client Channel's idle timeout and SharedResourceHolder's DESTROY_DELAY_SECONDS take care of a series of calls from a single client?

I'll come back to either remove this or justify it with a comment in a future PR.